src/game/Unit.cpp

```diff
@@ -11085,11 +11085,13 @@ void Unit::RemoveFromWorld()
 
 void Unit::CleanupsBeforeDelete()
 {
+    if (!IsInWorld())
+    return;
+
     // This needs to be before RemoveFromWorld to make GetCaster() return a valid pointer on aura removal
```

Comment:
See this line for why

Reply:
No need do something if unit already not in world.
All correct, test this patch work like a charm!